superset/models/core.py

```diff
@@ -665,6 +665,7 @@ def get_sqla_engine(self, schema=None, nullpool=True, user_name=None):
         logging.info('Database.get_sqla_engine(). Masked URL: {0}'.format(masked_url))
 
         params = extra.get('engine_params', {})
+        self.cursor_kwargs = self.db_engine_spec.get_cursor_configuration_for_impersonation(str(url), self.impersonate_user, effective_username)
```

Comment:
Stamping into `self.cursor_kwargs` here is brittle as it assumes method call ordering which isn't an explicit assumption. Either make this happen on the `__init__` method if the context is available at that point, or use a `@property`.